### PR TITLE
NIFI-10586 Prioritize ssh-rsa algorithm in SFTP Processors

### PR DESCRIPTION
**Pull request by exceptionfactory**
# Summary

[NIFI-10586](https://issues.apache.org/jira/browse/NIFI-10586) Prioritizes the `ssh-rsa` Key Algorithm in SFTP Processors when the `Key Algorithms Allowed` property is not configured.

This approach provides the widest degree of compatibility in the default configuration, following changes in SSHJ 0.33.0. The default SSHJ configuration supports `ssh-rsa` as well as `rsa-sha2-256` and `rsa-sha2-512`, but this change prioritizes the legacy algorithm for better compatibility. SSHJ 0.33.0 also supports trying all configured RSA algorithms, which allows SSHJ to communicate with servers that have disabled the `ssh-rsa` algorithm.
# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by thenatog** [comment]
Will review

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ssh/StandardSSHConfigProvider.java [hunk 1]
```diff
@@ -67,6 +67,13 @@ public Config getConfig(final String identifier, final PropertyContext context)
         getOptionalProperty(context, KEY_EXCHANGE_ALGORITHMS_ALLOWED).ifPresent(property -> config.setKeyExchangeFactories(getFilteredValues(property, config.getKeyExchangeFactories())));
         getOptionalProperty(context, MESSAGE_AUTHENTICATION_CODES_ALLOWED).ifPresent(property -> config.setMACFactories(getFilteredValues(property, config.getMACFactories())));
 
+        final String keyAlgorithmsAllowed = context.getProperty(KEY_ALGORITHMS_ALLOWED).evaluateAttributeExpressions().getValue();
+        if (keyAlgorithmsAllowed == null) {
+            // Prioritize ssh-rsa when Key Algorithms Allowed is not specified
+            config.prioritizeSshRsaKeyAlgorithm();
```
Comment (thenatog) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ssh/StandardSSHConfigProvider.java:73]: Why does this need to happen when keyAlgorithmsAllowed == null and then again on line 76? Should it only be happening here and not below?

Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ssh/StandardSSHConfigProvider.java:73]: Thanks for the catch, that should not be happening, will remove the unconditional call from line 76.

**Comment by thenatog** [comment]
+1 confirmed that the ssh-rsa algorithm is prioritized when prioritizeSshRsaKeyAlgorithm() is called both through the test and at run time. I noticed when debugging at run time that getClient() and thereby getConfig() gets called every processor execution and not just on initialization. Not sure how big a deal this is.

**Comment by exceptionfactory** [comment]
Thanks for the review @thenatog!

Some of the SFTP Processors call `getClient()` on every invocation as noted. Although this works, it is not optimal and there is a separate issue [NIFI-10450](https://issues.apache.org/jira/browse/NIFI-10450) to improve the behavior.

**Comment by thenatog** [comment]
Ah good to know it's already been noticed :)

Will merge.

**Pull request closed without merging**